agent/xds/routes.go

```diff
@@ -599,44 +611,43 @@ func makeUpstreamRouteForDiscoveryChain(
 		}
 
 	case structs.DiscoveryGraphNodeTypeSplitter:
-		routeAction, err := makeRouteActionForSplitter(startNode.Splits, chain, clusterNamePrefix)
-		if err != nil {
-			return nil, err
-		}
+		routeAction, err := s.makeRouteActionForSplitter(upstreamsSnapshot, startNode.Splits, chain, forMeshGateway)
+		if err == nil {
+			var lb *structs.LoadBalancer
+			if startNode.LoadBalancer != nil {
+				lb = startNode.LoadBalancer
+			}
+			if err := injectLBToRouteAction(lb, routeAction.Route); err != nil {
+				return nil, fmt.Errorf("failed to apply load balancer configuration to route action: %v", err)
+			}
 
-		var lb *structs.LoadBalancer
-		if startNode.LoadBalancer != nil {
-			lb = startNode.LoadBalancer
-		}
-		if err := injectLBToRouteAction(lb, routeAction.Route); err != nil {
-			return nil, fmt.Errorf("failed to apply load balancer configuration to route action: %v", err)
-		}
+			defaultRoute := &envoy_route_v3.Route{
+				Match:  makeDefaultRouteMatch(),
+				Action: routeAction,
+			}
 
-		defaultRoute := &envoy_route_v3.Route{
-			Match:  makeDefaultRouteMatch(),
-			Action: routeAction,
+			routes = []*envoy_route_v3.Route{defaultRoute}
 		}
 
-		routes = []*envoy_route_v3.Route{defaultRoute}
-
 	case structs.DiscoveryGraphNodeTypeResolver:
-		routeAction := makeRouteActionForChainCluster(startNode.Resolver.Target, chain, clusterNamePrefix)
+		routeAction, ok := s.makeRouteActionForChainCluster(upstreamsSnapshot, startNode.Resolver.Target, chain, forMeshGateway)
+		if ok {
```

Comment:
Could reduce the amount of indentation here by breaking if `!ok`
```
if !ok {
	break
}
```

Reply:
Ah, nice. I tried to use `continue` and it didn't work. This is awesome.